src/full/Agda/TypeChecking/Unquote.hs

```diff
@@ -753,6 +767,21 @@ evalTCM v = do
       else
         quoteTerm =<< process v
 
+    tcElaborate :: R.PostponedTerm -> R.Type -> TCM Term
+    tcElaborate (R.Postponed {..}) u = do
+      r <- isReconstructed
+      u <- workOnTypes $ locallyReduceAllDefs $ isType_ =<< toAbstract_ u
+      reportSDoc "tc.unquote.elaborate" 50 $ "Elaborating against Type:" <+> pretty u
+      equalType u (El (MetaS postponedSort []) (MetaV postponedType []))
+      e <- toAbstract_ postponedTerm
+      v <- checkExpr e u
+      if r then do
+        v <- process v
+        v <- locallyReduceAllDefs $ reconstructParameters u v
+        locallyReconstructed (quoteTerm v)
+      else
+        quoteTerm =<< process v
+
```

Comment:
This implementation is weird: you're checking the _metavariable_ which results from postponement against the provided type, even after solving `postponedType := u`, which causes the blocked type checking problem to resume. So there's some amount of duplicated work going on here!